guides/any_fixture.md

```diff
@@ -58,3 +58,78 @@ require "test_prof/recipes/rspec/any_fixture"
 ```
 
 Now you can use `TestProf::AnyFixture` in your tests.
+
+## Caveats
+
+`AnyFixture` cleans tables in the reversed order, in which they were registered. This
+means, that if you register a fixture, which references a not-yet-registed table, a
+foreign-key violation error *might* occur. An example is worth more than 1000
+words:
+
+```ruby
+class Author < ApplicationRecord
+  has_many :articles
+end
+
+class Article < ApplicationRecord
+  belongs_to :author
+end
+```
+
+The usual usage of would be:
+
+1. Register the shared contexts:
+
+```ruby
+RSpec.shared_context "author" do
+  before(:all) do
+    @author = TestProf::AnyFixture.register(:author) do
+      FactoryGirl.create(:account)
+    end
+  end
+
+  let(:author) { @author }
+end
+
+RSpec.shared_context "article" do
+  before(:all) do
+    @article = TestProf::AnyFixture.register(:article) do
+      FactoryGirl.create(:article, author: @author)
+    end
+  end
+
+  let(:article) { @article }
+end
+```
+
+2. Include the contexts into a spec file:
+
+```ruby
+include_context 'author'
+include_context 'article'
+```
+
+If one forgets to include the `author` context or includes it after the `article` context,
+an error will be raised. At the end of the suite, first the `articles` and then the
```

Comment:
That's not true, actually. If we don't include the "author" context (or include it after) there is no `@author` during the execution of the "article" context, so it's `nil`.

Then either `create(:article, author: nil)` fails (if there is a validation) or nothing bad happens, 'cause `@article` wouldn't have a `author_id`.

Reply:
@palkan I will check tomorrow, but if I am not mistaken, if I refence `@author`, which doesn't exist yet, I get an error (maybe because in Rails 5 belongs_to is required by default, however, I don't remember what the error exactly was). I'll check tomorrow and I'll make a new PR.

Reply:
What you say sounds quite logical.

Reply:
Yep, Rails 5 `belongs_to` adds `required: true` by default.

I've updated the guide and left the second example only.